include/action.h

```diff
@@ -2,19 +2,48 @@
 #ifndef __LABWC_ACTION_H
 #define __LABWC_ACTION_H
 
-struct server;
+/* Forward declare arguments */
 struct view;
+struct server;
+struct wl_list;
+struct wlr_box;
 
 struct action {
-	uint32_t type;
-	char *arg;
-	struct wl_list link;
+	struct wl_list link;  /* struct keybinding.actions,
+	                       * struct mousebinding.actions,
+	                       * struct menuitem.actions */
+
+	uint32_t type;        /* enum action_type */
+	struct wl_list args;  /* struct action_arg.link */
 };
 
-struct action *action_create(const char *action_name);
-void action_list_free(struct wl_list *action_list);
+struct action_arg {
+	struct wl_list link;  /* struct action.args */
 
+	/**
+	 * FIXME: Do we want to replace `name` by a plain uint32_t and have
+	 * an enum for each Action which uses more than a single argument?
+	 * That would allow to not constantly strcmp() strings in Action handlers.
+	 *
+	 * Alternative:
+	 * For some Actions we could also have a struct like action_arg_box
+	 * which derives from struct action_arg and then carries around random
+	 * things like a wlr_box which gets filled while parsing the
+	 * key/mouse/menu binds. `MoveResizeTo` comes to mind :)
+	 */
+	const char *name;  /* May be NULL if there is just one arg */
+	uint32_t type;     /* enum action_arg_type */
+};
+
+
+struct action *action_create(const char *action_name);
+void action_add_str_arg(struct action *action, char *name, const char *value);
+void action_add_int_arg(struct action *action, char *name, int value);
+void action_add_double_arg(struct action *action, char *name, double value);
+void action_add_bool_arg(struct action *action, char *name, bool value);
+struct wlr_box *action_add_box_arg(struct action *action, char *name);
```

Comment:
Only `action_add_str_arg()` is used in this PR, the others are examples of how this action_arg framework could be used